### PR TITLE
Swap snark-verifier for halo2-solidity-verifier

### PR DESCRIPTION
**Pull request by alxkzmn**
Here's the difference between our fork of halo2-solidity-verifier and the default version:
https://github.com/privacy-scaling-explorations/halo2-solidity-verifier/compare/main...summa-dev:halo2-solidity-verifier:main

### CODE REVIEW

**Review by enricobottazzi — APPROVED** [review]
Left minor comments. On top of the PR, I wonder if this new `halo2_solidity_verifier` somehow reduced the size of the deployed contracts. Were you able to benchmark this?

File: zk_prover/Cargo.toml [hunk 1]
```diff
@@ -10,8 +10,8 @@ dev-graph = ["halo2_proofs/dev-graph", "plotters"]
 
 
 [dependencies]
-halo2_proofs = { git = "https://github.com/privacy-scaling-explorations/halo2", tag = "v2023_04_20"}
-halo2_gadgets = { git = "https://github.com/privacy-scaling-explorations/halo2", tag = "v2023_04_20"}
+halo2_proofs = { git = "https://github.com/summa-dev/halo2"}
```
[the remark this answers is not in the document]
Comment (alxkzmn) [reply zk_prover/Cargo.toml:13]: I was thinking about making a PR but I need to discuss one of the changes I made with Han.

**Comment by alxkzmn** [comment]
@enricobottazzi the verifier deployment gas usage has reduced a bit from 4822482 to 4228615 (by ~600000), so I guess the bytecode is also smaller.

File: zk_prover/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (alxkzmn) [reply zk_prover/Cargo.toml:13]: I'll explain the fork in the v2 readme.

File: contracts/package.json [hunk 2]
```diff
@@ -8,10 +8,10 @@
   "devDependencies": {
     "@nomicfoundation/hardhat-network-helpers": "^1.0.8",
     "@nomicfoundation/hardhat-toolbox": "^2.0.2",
-    "hardhat": "^2.16.0",
```
Comment (sifnoc) [inline contracts/package.json:11]: I wasn't sure where to leave a comment about the README, so I'm posting it here.

It seems that running `npx hardhat node` is no longer a prerequisite for executing `npx hardhat test` when testing contracts. Could we consider moving the `npx hardhat node` command line to the Deploying section in the README?

https://github.com/summa-dev/summa-solvency/tree/halo2-verifier/contracts#testing

**Review by sifnoc — CHANGES_REQUESTED** [review]

**Review by sifnoc — APPROVED** [review]

**Pull request merged**